plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1157,6 +1158,24 @@ else if (avroSchemaLiteral != null) {
         tableProperties.put("numFiles", "-1");
         tableProperties.put("totalSize", "-1");
 
+        // Extra properties
+        getExtraProperties(tableMetadata.getProperties())
+                .ifPresent(extraProperties -> {
+                    Set<String> existingProperties = tableProperties.buildOrThrow().keySet();
+
+                    if (existingProperties.stream().anyMatch(extraProperties::containsKey)) {
+                        throw new TrinoException(
+                                INVALID_TABLE_PROPERTY,
+                                "Invalid keys in extra_properties: %s".formatted(
+                                        extraProperties.keySet().stream()
+                                                .filter(existingProperties::contains)
+                                                .collect(joining(" ,"))));
+                    }
+                    for (Map.Entry<String, String> extraProperty : extraProperties.entrySet()) {
+                        tableProperties.put(extraProperty.getKey(), extraProperty.getValue());
```

Comment:
What about these properties set by Trino later down the road
https://github.com/trinodb/trino/blob/5dc41d458be2af705a8433058efe2f33254ce572/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java#L777-L780
?

Reply:
Have exposed `STATS_PROPERTIES` in `HiveMetadata` - but not sure if it is the correct approach as `HiveMetadata` is independent of the type of metastore like thrift or glue but not we directly import from thrift package. Any workarounds or pointer would be helpful.